src/compiler/checker.ts

```diff
@@ -5341,37 +5341,53 @@ namespace ts {
                 if (isArrayType(type)) {
                     return createArrayType(getWidenedType((<TypeReference>type).typeArguments[0]));
                 }
+                if (isTupleType(type)) {
+                    return createTupleType(map(type.elementTypes, getWidenedType));
+                }
             }
             return type;
         }
 
-        function reportWideningErrorsInType(type: Type): boolean {
+        function reportWideningErrorsInType(type: Type, expression: Expression): boolean {

```

Comment:
I like idea of rename the ```expression``` to ```expressionForError```.  Also it will be helpful to make jsdoc comment documenting the function particularly when it will not report an error